app/storage_strategies/aws_s3.py

```diff
@@ -0,0 +1,64 @@
+import boto3
+import os
+from string import Template
+from botocore.exceptions import NoCredentialsError, ClientError
+from storage_strategies.storage_strategy import StorageStrategy
+
+class AWSS3StorageStrategy(StorageStrategy):
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.bucket_name = self._resolve_placeholder(context['settings'].get('bucket_name'))
+        self.region = self._resolve_placeholder(context['settings'].get('region'))
+        self.access_key = self._resolve_placeholder(context['settings'].get('access_key'))
+        self.secret_access_key = self._resolve_placeholder(context['settings'].get('secret_access_key'))
+
+        self.s3_client = boto3.client(
+            's3',
+            aws_access_key_id=self.access_key,
+            aws_secret_access_key=self.secret_access_key,
+            region_name=self.region
+        )
+
+    def _resolve_placeholder(self, value):
+        if not value:
+            return None
+        return Template(value).substitute(os.environ)
+
+    def save(self, file_name, dest_file_name, content):
+        formatted_file_name = self.format_file_name(file_name, dest_file_name)
+
+        try:
+            self.s3_client.put_object(
+                Bucket=self.bucket_name,
+                Key=formatted_file_name,
+                Body=content.encode('utf-8')
+            )
+            print(f"File {formatted_file_name} saved to S3 bucket {self.bucket_name}.")
+        except NoCredentialsError:
```

Comment:
In the other strategies errors are not handled on this level - raised up and handled in the storage_manager - API

I think it should be the way (not handling exception here but leaving it to level up) to make sure they are always resolved the same way

Wdyt?

Reply:
I think we should handle exceptions specific to the source libraries used in the storage_strategies and re-throw them as a common exception to ensure consistent error handling. Sometimes, we may need to add specific information to an exception, and this should be done at the strategy level, not in the storage_manager. What do you think about this approach? If it's acceptable, I will re-throw the exceptions. If not, I will implement it as you suggest.

Reply:
Fair point. Please handle and re-throw

Reply:
Ale ta dyskusja spowodowała, że nieco zluzowałem z exceptionami, których i tak sensownie nie wzbogacę - przerzuciłem sprawdzenie połączenia na inita, a same metody rethrowują wzbogacony exception aby pomoc w rozpoznaniu problemu jak np.
<img width="750" alt="image" src="https://github.com/user-attachments/assets/b09affc6-0d78-4502-a6cb-b3647dc878c6">
<img width="752" alt="image" src="https://github.com/user-attachments/assets/c2682633-7723-4032-bfab-fe2728863787">